File: sdks/csharp/PgChangeFeed.Client/Http/Models/ErrorResponse.cs
using System.Text.Json.Serialization;

namespace PgChangeFeed.Client.Http.Models;

/// <summary>
/// The error body of every non-success response
/// (<c>{"error": "&lt;text&gt;"}</c>) — internal, because it is only ever used
/// to build a typed <see cref="PgChangeFeedException"/>; it is never a return
/// value of a public method.
/// </summary>
internal sealed record ErrorResponse(
    [property: JsonPropertyName("error")] string Error);

File: sdks/kotlin/pgchangefeed-kotlin/src/main/kotlin/io/github/pt9912/pgchangefeed/http/model/ErrorResponse.kt
package io.github.pt9912.pgchangefeed.http.model

import com.google.gson.annotations.SerializedName

/**
 * The error body of every non-success response (`{"error": "<text>"}`) —
 * internal, because it is only ever used to build a typed
 * `PgChangeFeedException`; it is never a return value of a public method.
 */
internal data class ErrorResponse(
    @SerializedName("error") val error: String,
)
